src/windows/cppwinrt/LottieReactNative.idl

```diff
@@ -0,0 +1,43 @@
+namespace LottieReactNative
+{
+    interface ILottieMediaSource
+    {
+        Double FrameToProgress(Double frame);
+    };
+
+    interface ILottieSourceProvider
+    {
+        Windows.Foundation.IAsyncOperation<Microsoft.UI.Xaml.Controls.IAnimatedVisualSource> GetSourceFromName(String name);
+        Windows.Foundation.IAsyncOperation<Microsoft.UI.Xaml.Controls.IAnimatedVisualSource> GetSourceFromJson(String json);
+    }
+
+    runtimeclass ReactPackageProvider : [default]Microsoft.ReactNative.IReactPackageProvider
+    {
+        ReactPackageProvider(ILottieSourceProvider sourceProvider);
+    }
+
+    //
+    // Subclassing WinUI 2.x classes is broken, so we'll subclass Grid instead and compose
+    //
+    [default_interface]
+    runtimeclass LottieView : Windows.UI.Xaml.Controls.Grid
+    {
+        LottieView();
+        LottieView(Microsoft.ReactNative.IReactContext context, ILottieSourceProvider sourceProvider);
+
+        Windows.UI.Xaml.Media.Stretch ResizeMode { get; set; };
+
+        void SetSourceName(String name);
+        void SetSourceJson(String json);
+        void SetLoop(Boolean loop);
+        void SetProgress(Double progress);
+        void SetSpeed(Double speed);
```

Comment:
Most aren't applied immediately, but instead queued up to take effect at varying times based on load state. As such, its not clear what you'd want as the read value -- simply what was last passed in, or the actual state at that moment? Since there's nothing that needs to read them right now, I figured any decision there could be made when there's a scenario that needs the info.